test/libsolidity/MemoryGuardTest.cpp

```diff
@@ -58,13 +61,14 @@ TestCase::TestResult MemoryGuardTest::run(std::ostream& _stream, std::string con
 		ErrorList errors;
 		auto [object, analysisInfo] = yul::test::parse(
 			compiler().yulIR(contractName),
-			EVMDialect::strictAssemblyForEVMObjects({}),
+			EVMDialect::strictAssemblyForEVMObjects(CommonOptions::get().evmVersion()),
```

Comment:
Haha, I assumed we had no memory guards tests with anything other than the default version then?

Reply:
Yeah, we didn't. I don't think it's dependent on EVM version in a significant way though. Even now, I only had to change it because this preparatory Yul parsing step won't parse `mcopy` on earlier EVMs and not because of anything specifically related to the memory guard.